server/src/main/scala/net/wiringbits/repositories/daos/DatabaseTablesDAO.scala

```diff
@@ -19,9 +19,16 @@ object DatabaseTablesDAO {
       """.as(tableParser.*)
   }
 
-  def getTableMetadata(tableName: String)(implicit conn: Connection): TableMetadata = {
-    val statement = conn.createStatement(ResultSet.CONCUR_READ_ONLY, ResultSet.CONCUR_READ_ONLY)
-    val resultSet = statement.executeQuery("SELECT * FROM " + tableName)
+  def getTableMetadata(tableName: String, pagination: Pagination)(implicit conn: Connection): TableMetadata = {
+    // I still haven't found how to use preparedStatement with tableName
+    val sql = f"SELECT * FROM $tableName LIMIT ? OFFSET ?"
```

Comment:
You are querying the whole table just to find the metadata, this doesn't need pagination, just put `LIMIT 0` or `LIMIT 1` if `0` doesn't work.

Also, I see you use this for retrieving the data, `OFFSET` is useless unless you define a deterministic ordering